website/blog/2025-07-03-streamlining-form-management-heyform-ferretdb.md

```diff
@@ -0,0 +1,241 @@
+- **Data management:** Collect, view, and export submission data.
+- **Self-hostable:** Full control over your data and infrastructure.
+
+HeyForm focuses on providing a flexible and developer-friendly platform that helps you gather information efficiently and integrate it into your applications.
+
+## Why use HeyForm with FerretDB?
+
+HeyForm uses MongoDB as its primary database backend for storing form definitions, submissions, and user data.
+Given that FerretDB is designed to be a truly open source alternative to MongoDB, it can serve as a drop-in replacement for HeyForm's database.
+This powerful combination offers several compelling advantages:
+
+- **Open-source:** Both HeyForm and FerretDB are open-source projects, providing transparency, flexibility, and strong community backing, aligning perfectly with an open-source ethos.
+- **Simplified infrastructure:** If your existing data infrastructure is already based on PostgreSQL, integrating HeyForm with FerretDB can streamline your database management and reduce operational overhead.
+- **No vendor lock-in:** Enjoy the freedom of truly open-source solutions without concerns about proprietary licensing or vendor lock-in.
+
+## Connecting HeyForm to FerretDB
+
+Connecting HeyForm to your FerretDB instance is straightforward, as HeyForm uses MongoDB for primary data storage and KeyDB for caching and session management.
+Refer to the [HeyForm self-hosting documentation](https://docs.heyform.net/open-source/self-hosting) for more details on its configuration options.
+
+Here's a step-by-step guide to get you started:
+
+1. **Setup Docker Compose file:**
+   Create a `docker-compose.yml` file with the following content to define the services for HeyForm, FerretDB, and KeyDB (for caching).
+
+   ```yaml
+   services:
+     heyform:
+       image: heyform/community-edition:latest
+       restart: always
+       volumes:
+         # Persist uploaded images
+         - ./assets:/app/static/upload
+       depends_on:
+         - ferretdb
+         - keydb
+       ports:
+         - '9513:8000'
+       environment:
+         APP_HOMEPAGE_URL: http://127.0.0.1:9513
+         SESSION_KEY: key1
+         FORM_ENCRYPTION_KEY: key2
+         MONGO_URI: 'mongodb://<username>:<password>@ferretdb:27017/heyform'
+         REDIS_HOST: keydb
+         REDIS_PORT: 6379
+
+     ferretdb:
+       image: ghcr.io/ferretdb/ferretdb-eval:2
+       restart: on-failure
+       ports:
+         - 27017:27017
+       environment:
+         - POSTGRES_USER=<username>
+         - POSTGRES_PASSWORD=<password>
+         - POSTGRES_DB=postgres
+       volumes:
+         # Persist FerretDB data
+         - ferret_data:/var/lib/postgresql/data
+
+     keydb:
+       image: eqalpha/keydb
+       restart: always
+       command: keydb-server --appendonly yes --protected-mode no
+       volumes:
+         # Persist KeyDB data
+         - keydb:/data
+
+   volumes:
+     ferret_data:
```

Comment:
Let's use `ferretdb_data`. Also Let's use `ferretdb` instead of `ferret` in other places too.